huaweicloud/services/cph/resource_huaweicloud_cph_adb_command.go

```diff
@@ -0,0 +1,213 @@
+	if len(phoneIds) > 0 {
+		log.Printf("[WARN] Phone not found: %v", phoneIds)
+	}
+	if len(serverIds) > 0 {
+		log.Printf("[WARN] Parameter: server_id is invalid: %v", serverIds)
+	}
+
+	err = checkAdbCommandJobStatus(ctx, client, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary:  fmt.Sprint(err),
+			},
+		}
+	}
+
+	return nil
+}
+
+func buildcreateAdbCommandBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"command":    d.Get("command"),
+		"content":    d.Get("content"),
+		"phone_ids":  utils.ValueIgnoreEmpty(d.Get("phone_ids")),
+		"server_ids": utils.ValueIgnoreEmpty(d.Get("server_ids")),
+	}
+
+	return bodyParams
+}
+
+func resourceAdbCommandRead(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceAdbCommandUpdate(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceAdbCommandDelete(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	errorMsg := "Deleting adb command resource is not supported. The resource is only removed from the state."
+	return diag.Diagnostics{
+		diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  errorMsg,
+		},
+	}
+}
+
+func checkAdbCommandJobStatus(ctx context.Context, client *golangsdk.ServiceClient, id string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      jobStateRefreshFunc(client, id),
+		Timeout:      timeout,
+		PollInterval: 10 * timeout,
+		Delay:        10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for CPH adb job completed: %s", err)
+	}
+	return nil
+}
+
+func jobStateRefreshFunc(client *golangsdk.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		getJobsHttpUrl := "v1/{project_id}/cloud-phone/jobs?request_id={request_id}"
+		getJobsPath := client.Endpoint + getJobsHttpUrl
+		getJobsPath = strings.ReplaceAll(getJobsPath, "{project_id}", client.ProjectID)
+		getJobsPath = strings.ReplaceAll(getJobsPath, "{request_id}", id)
+
+		getJobsOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+
+		getJobsResp, err := client.Request("GET", getJobsPath, &getJobsOpt)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		getJobsRespBody, err := utils.FlattenResponse(getJobsResp)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		runningStatus := utils.PathSearch("jobs[?status==`1`]", getJobsRespBody, make([]interface{}, 0)).([]interface{})
+		failedPhoneIds := utils.PathSearch("jobs[?status==`-1`].phone_id", getJobsRespBody, make([]interface{}, 0)).([]interface{})
```

Comment:
It is recommended to add corresponding comments for status value.

Reply:
![image](https://github.com/user-attachments/assets/58c665eb-d84a-427a-af29-6ed7ab9d3fd8)